### PR TITLE
Add `ops.map_coordinates`

### PR DESCRIPTION
**Pull request by james77777778**
Related to #481 

This PR has implemented `ops.map_coordinates` for all backends based on the PR from @mihirparadkar #784 

It is challenge to obtain a jittable `map_coordinates` for tensorflow, but I managed to figure out the solution. The key is to use `tf.unstack` to separate coordinates and form a list of tensor for subsequent operations.

The unit test is borrowed from jax and has been simpified
https://github.com/google/jax/blob/bcc545a69232e983ae31b0395f4972979f2789c0/tests/scipy_ndimage_test.py#L79

The standalone script:
```python
import math

import numpy as np

from keras_core.backend.jax.image import map_coordinates as jax_map_coordinates
from keras_core.backend.numpy.image import map_coordinates as np_map_coordinates
from keras_core.backend.tensorflow.image import map_coordinates as tf_map_coordinates
from keras_core.backend.torch.image import map_coordinates as torch_map_coordinates
import tensorflow as tf

np.random.seed(42)
shape = (3, 4, 5)
coords_shape = (2, 3, 4)
dtype = "float32"
x = np.arange(math.prod(shape), dtype=dtype).reshape(shape)
coords = [
    (size - 1) * np.random.uniform(size=coords_shape).astype(dtype)
    for size in shape
]

print("jax:")
print(jax_map_coordinates(x, coords, 1))
print("np:")
print(np_map_coordinates(x, coords, 1))
print("torch:")
print(torch_map_coordinates(x, coords, 1))
print("tf eager:")
print(tf_map_coordinates(x, coords, 1))
print("tf xla:")
print(tf.function(tf_map_coordinates, jit_compile=True)(x, coords, 1))

```

Results:

<details>

```bash
Using TensorFlow backend
jax:
[[[24.009495  50.545628  36.153202  34.760387 ]
  [18.884958  10.515846  13.828117  40.892403 ]
  [25.374344  43.34012   15.488769  52.22368  ]]

 [[39.421623  11.044044  20.851446  15.36548  ]
  [15.1240015 30.588694  18.357327  28.497757 ]
  [28.654016  19.465136  19.45043   23.250359 ]]]
np:
[[[24.009495  50.54563   36.153202  34.76039  ]
  [18.884958  10.515847  13.828115  40.892403 ]
  [25.374344  43.340122  15.488769  52.22368  ]]

 [[39.42162   11.044042  20.851444  15.36548  ]
  [15.1240015 30.588696  18.357325  28.497759 ]
  [28.654016  19.465137  19.450432  23.250357 ]]]
torch:
tensor([[[24.0095, 50.5456, 36.1532, 34.7604],
         [18.8850, 10.5158, 13.8281, 40.8924],
         [25.3743, 43.3401, 15.4888, 52.2237]],

        [[39.4216, 11.0440, 20.8514, 15.3655],
         [15.1240, 30.5887, 18.3573, 28.4978],
         [28.6540, 19.4651, 19.4504, 23.2504]]], device='cuda:0')
tf eager:
tf.Tensor(
[[[24.009495  50.545628  36.153202  34.760387 ]
  [18.884958  10.515846  13.828117  40.892403 ]
  [25.374344  43.34012   15.488769  52.22368  ]]

 [[39.421623  11.044044  20.851446  15.36548  ]
  [15.1240015 30.588694  18.357327  28.497757 ]
  [28.654016  19.465136  19.45043   23.250359 ]]], shape=(2, 3, 4), dtype=float32)
tf xla:
tf.Tensor(
[[[24.009495  50.545628  36.153202  34.760387 ]
  [18.884958  10.515846  13.828117  40.892403 ]
  [25.374344  43.34012   15.488769  52.22368  ]]

 [[39.421623  11.044044  20.851446  15.36548  ]
  [15.1240015 30.588694  18.357327  28.497757 ]
  [28.654016  19.465136  19.45043   23.250359 ]]], shape=(2, 3, 4), dtype=float32)
```

</details>

### CODE REVIEW

File: keras_core/ops/image.py [hunk 1]
```diff
@@ -418,3 +418,90 @@ def _extract_patches(
...
+
+@keras_core_export("keras_core.ops.image.map_coordinates")
+def map_coordinates(
+    input, coordinates, order, fill_mode="constant", fill_value=0
+):
+    """Map the input array to new coordinates by interpolation..
+
+    Note that interpolation near boundaries differs from the scipy function,
+    because we fixed an outstanding bug
+    https://github.com/scipy/scipy/issues/2640.
```
Comment (fchollet) [inline keras_core/ops/image.py:463]: Please use markdown for links.

**Review by fchollet — COMMENTED** [review]
Thanks for the PR -- Excellent work! 👍

File: keras_core/ops/image.py (hunk 1 above)
Comment (james77777778) [reply keras_core/ops/image.py]: Fixed

**Review by fchollet — APPROVED** [review]
Thank you for the great contribution!

**Pull request merged**